rdtools/soiling.py

```diff
@@ -0,0 +1,987 @@
+'''
+Functions for calculating soiling metrics from photovoltaic system data.
+
+The soiling module is currently experimental. The API, results,
+and default behaviors may change in future releases (including MINOR
+and PATCH releases) as the code matures.
+'''
+
+import warnings
+import pandas as pd
+import numpy as np
+from scipy.stats.mstats import theilslopes
+
+warnings.warn(
+    'The soiling module is currently experimental. The API, results, '
+    'and default behaviors may change in future releases (including MINOR '
+    'and PATCH releases) as the code matures.'
+)
+
+# Custom exception
+class NoValidIntervalError(Exception):
+    '''raised when no valid rows appear in the result dataframe'''
+    pass
+
+
+class SRRAnalysis():
+    '''
+    Class for running the stochastic rate and recovery (SRR) photovoltaic
+    soiling loss analysis presented in Deceglie et al. JPV 8(2) p547 2018
```

Comment:
worth including the full reference somewhere. It's easier to search with a title.